br/pkg/restore/client_test.go

```diff
@@ -330,12 +331,35 @@ func TestPreCheckTableClusterIndex(t *testing.T) {
 type fakePDClient struct {
 	pd.Client
 	stores []*metapb.Store
+
+	notLeader bool
 }
 
 func (fpdc fakePDClient) GetAllStores(context.Context, ...pd.GetStoreOption) ([]*metapb.Store, error) {
 	return append([]*metapb.Store{}, fpdc.stores...), nil
 }
 
+func (fpdc fakePDClient) GetTS(ctx context.Context) (int64, int64, error) {
+	if fpdc.notLeader {
+		return 0, 0, errors.Errorf("rpc error: code = Unknown desc = [PD:tso:ErrGenerateTimestamp]generate timestamp failed, requested pd is not leader of cluster")
+	}
+	return 1, 1, nil
+}
+
+func TestGetTSWithRetry(t *testing.T) {
+	t.Log("case 1:")
```

Comment:
Perhaps using [`t.Run`](https://pkg.go.dev/testing@go1.19.2#T.Run) to start sub-cases.